src/Resizer/CropResizer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Resizer;
+
+use Gaufrette\File;
+use Imagine\Image\Box;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Imagine\Image\Point;
+use Sonata\MediaBundle\Metadata\MetadataBuilderInterface;
+use Sonata\MediaBundle\Model\MediaInterface;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class CropResizer implements ResizerInterface
+{
+    /**
+     * @var ImagineInterface
+     */
+    private $adapter;
+
+    /**
+     * @var MetadataBuilderInterface
+     */
+    private $metadata;
+
+    public function __construct(ImagineInterface $adapter, MetadataBuilderInterface $metadata)
+    {
+        $this->adapter = $adapter;
+        $this->metadata = $metadata;
+    }
+
+    public function resize(MediaInterface $media, File $in, File $out, $format, array $settings): void
+    {
+        if (!isset($settings['width'])) {
+            throw new \RuntimeException(sprintf(
+                'Width parameter is missing in context "%s" for provider "%s"',
+                $media->getContext(),
+                $media->getProviderName()
+            ));
+        }
+
+        if (!isset($settings['height'])) {
+            throw new \RuntimeException(sprintf(
+                'Height parameter is missing in context "%s" for provider "%s"',
+                $media->getContext(),
+                $media->getProviderName()
+            ));
+        }
+
+        $image = $this->adapter->load($in->getContent());
+
+        $sourceSize = $media->getBox();
+        $targetSize = $this->createTargetBox($settings);
+
+        if ($this->shouldModify($sourceSize, $targetSize)) {
+            $image = $this->cropImage($image, $sourceSize, $targetSize);
+        }
+
+        // Always change format and quality
+        $content = $image->get($format, [
+            'quality' => $settings['quality'],
+        ]);
+
+        $out->setContent($content, $this->metadata->get($media, $out->getName()));
+    }
+
+    public function getBox(MediaInterface $media, array $settings)
```

Comment:
Could we provide the type declaration?

Reply:
Not related to this PR. The `getBox` method is declared by the interface

Reply:
Adding return typehint is allowed, isn't it ?

Reply:
Done

Reply:
I don't see the return type declaration here.

Reply:
Sorry, I added it to the wrong class. Should be fixed now.